CONTRIBUTING.md

```diff
@@ -418,191 +351,1122 @@ This is called the `virtual path`. In this case it should be `stocks.dps.shorted
+
+This is the convention in use for creating a new key/value pair:
+
+- `stocks/search` - Under `stocks` context, short command `search` description on the `help menu`
+- `stocks/SEARCH` - Under `stocks` context, long command `search` description, when `search -h`
+- `stocks/SEARCH_query` - Under `stocks` context, `query` description when inquiring about `search` command with `search -h`
+- `stocks/_ticker` - Under `stocks` context, `_ticker` is used as a key of a parameter, and the displayed parameter description is given as value
+- `crypto/dd/_tokenomics_` - Under `crypto` context and under `dd` menu, `_tokenomics_` is used as a key of an additional information, and the displayed information is given as value
+
+TO BE ADDED - WORK IN PROGRESS
+
+## Write Code and Commit
+
+At this stage it is assumed that you have already forked the project and are ready to start working.
+
+### Pre Commit Hooks
+
+Git hook scripts are useful for identifying simple issues before submission to code review. We run our hooks on every
+commit to automatically point out issues in code such as missing semicolons, trailing whitespace, and debug statements.
+By pointing these issues out before code review, this allows a code reviewer to focus on the architecture of a change
+while not wasting time with trivial style nitpicks.
+
+Install the pre-commit hooks by running: `pre-commit install`.
+
+### Coding
+
+Although the Coding Guidelines section has been already explained. It is worth mentioning that if you want to be faster
+at developing a new feature, you may implement it first on a `jupyter notebook` and then carry it across to the
+terminal. This is mostly useful when the feature relies on scraping data from a website, or implementing a Neural
+Network model.
+
+### Git Process
+
+1. Create your Feature Branch, e.g. `git checkout -b feature/AmazingFeature`
+2. Check the files you have touched using `git status`
+3. Stage the files you want to commit, e.g.
+   `git add openbb_terminal/stocks/stocks_controller.py openbb_terminal/stocks/stocks_helper.py`.
+   Note: **DON'T** add `config_terminal.py` or `.env` files with personal information, or even `feature_flags.py` which is user-dependent.
+4. Write a concise commit message under 50 characters, e.g. `git commit -m "meaningful commit message"`. If your PR
+   solves an issue raised by a user, you may specify such issue by adding #ISSUE_NUMBER to the commit message, so that
+   these get linked. Note: If you installed pre-commit hooks and one of the formatters re-formats your code, you'll need
+   to go back to step 3 to add these.
+
+## Add a Test
+
+Unit tests minimize errors in code and quickly find errors when they do arise.
+
+### Pytest
+
+Pytest allows users to quickly create unit tests in Python. To use pytest run `pytest tests/`.
```

Comment:
@Chavithra I still find tests one of the hardest things to do on the terminal - and have gotten that feedback multiple times. Can we write more and better documentation around it?

Reply:
@DidierRLopes Got the same feelings. Chavi and I talked about this and I added some more explanations on tests a while ago to the README for tests. We could link it here as it's such an extensive topic

Reply:
Am fine with that. But definitely think we need to drill more on that topic. A lot of devs are a bit scared by tests.